### ISSUE TITLE
Questions

### ISSUE THREAD

**Issue by brecht-vermeersch** [Azure-OSS/azure-storage-php#29]
Hi. I am willing to help. 

The only problem I have is that the scope of the project is quite large and the only thing I need is the blob storage. Maybe we can remove everything that is not blob storage right now, and add it back later if we find somebody that is willing to maintain it. The large scope of the project probably scares of willing contributors.

**Comment by pimjansen** [comment]
Uhmm well i understand what you mean. Any suggestions on how to attack this?

The lib is clearly not build by a php dev and really messy

**Comment by brecht-vermeersch** [comment]
We could turn each package into a separate composer package. A bit like how laravel/framework is setup. 

Then we can add a path filter in the github actions so that the tests for Blob only run if there are changes in Blob (or common).

This could help us focus on getting Blob to a v1, even if the other packages are not ready yet.

**Comment by pimjansen** [comment]
> We could turn each package into a separate composer package. A bit like how laravel/framework is setup. 
> 
> 
> 
> Then we can add a path filter in the github actions so that the tests for Blob only run if there are changes in Blob (or common).
> 
> 
> 
> This could help us focus on getting Blob to a v1, even if the other packages are not ready yet.
> 
> 

I'd rather push it to a single repo since it removes overhead. In general it is a single service from Azure you consume. I do not mind having it not fully complete.

But we need to start splitting stuff and setting proper CI. I will try to do a setup this evening to see if we can continue from there.

Would that work?

**Comment by brecht-vermeersch** [comment]
Yes keeping it all in the same repo is easier. I already started on reorganising the tests to group them per package. I'm also trying to run the tests with azurite. I'll keep you updated on my progress

**Comment by pimjansen** [comment]
> Yes keeping it all in the same repo is easier. I already started on reorganising the tests to group them per package. I'm also trying to run the tests with azurite. I'll keep you updated on my progress

I will be on the slack channel later this evening:

Join me on Slack -- it’s a faster, simpler way to work. Sign up here, from any device: https://join.slack.com/t/azure-oss/shared_invite/zt-2libh4fcl-6pRDd2aUXeICeagZf1JeFw
